contracts/utils/math/Math.sol

```diff
@@ -339,38 +339,113 @@ library Math {
      * @dev Returns the square root of a number. If the number is not a perfect square, the value is rounded
      * towards zero.
      *
-     * Inspired by Henry S. Warren, Jr.'s "Hacker's Delight" (Chapter 11).
+     * This method is based on Newton's method for computing square roots; the algorithm is restricted to only
```

Comment:
> f I understand correctly, this is more based on Heron's method than Newton's. I honestly don't see a reason to reference Newton's method if Heron's is more specific.

Newton's method is a mathematical approach that is very well known and that applies to many function. Heron's method is the name given to Newton's method in the case of one specific function. Heron's method was known long before Newton's method generalized te approach. Most people don't know "Heron's" method by name.

If you think mentionning the two is confusing, than I'd just keep Newton's because that is the more generic and widelly documented of the two.

To make a analogy:
- Newton's method → ECDSA
- Heron's method → Secp256k1

Reply:
I thought we'd be interested in getting the most specific version. But you're right on the search result.

This would justify mentioning both.